src/Widget/FormBuilder.php

```diff
@@ -0,0 +1,385 @@
+use Yiisoft\Widget\Widget;
+
+final class FormBuilder extends Widget
+{
+    use Collection\Options;
+
+    public const VALIDATION_STATE_ON_CONTAINER = 'container';
+    public const VALIDATION_STATE_ON_INPUT = 'input';
+    private string $action = '';
+    private string $method = Method::POST;
+    private bool $encodeErrorSummary = true;
+    private string $errorCssClass = 'has-error';
+    private string $errorSummaryCssClass = 'error-summary';
+    private string $inputCssClass = 'form-control';
+    private string $requiredCssClass = 'required';
+    private string $successCssClass = 'has-success';
+    private string $validatingCssClass = 'validating';
+    private string $validationStateOn = self::VALIDATION_STATE_ON_CONTAINER;
+    private ?string $validationUrl = null;
+    private bool $validateOnSubmit = true;
+    private bool $validateOnChange = true;
+    private bool $validateOnBlur = true;
+    private bool $validateOnType = false;
+    private int $validationDelay = 500;
+    private string $ajaxParam = 'ajax';
+    private string $ajaxDataType = 'json';
+    private bool $scrollToError = true;
+    private int $scrollToErrorOffset = 0;
+    private array $attributes = [];
+
+    public function start(): self
+    {
+        ob_start();
+        ob_implicit_flush(0);
+
+        return $this;
+    }
+
+    public function run(): string
+    {
+        $content = ob_get_clean();
+
+        $html = Forms::begin()
+            ->action($this->action)
+            ->method($this->method)
+            ->options($this->options)
+            ->start();
+        $html .= $content;
+        $html .= Forms::end();
+
+        return $html;
+    }
+
+    /**
+     * Generates a summary of the validation errors.
+     *
+     * If there is no validation error, an empty error summary markup will still be generated, but it will be hidden.
+     *
+     * @param FormInterface $forms the forms(s) associated with this form.
+     * @param array $options the tag options in terms of name-value pairs. The following options are specially handled:
+     *
+     * - `header`: string, the header HTML for the error summary. If not set, a default prompt string will be used.
+     * - `footer`: string, the footer HTML for the error summary.
+     *
+     * The rest of the options will be rendered as the attributes of the container tag. The values will be HTML-encoded
+     * using {@see \Yiisoft\Html\Html::encode()}. If a value is `null`, the corresponding attribute will not be
+     * rendered.
+     *
+     * @throws InvalidConfigException
+     *
+     * @return string the generated error summary.
+     *
+     * {@see errorSummaryCssClass}
+     */
+    public function errorSummary(FormInterface $forms, array $options = []): string
+    {
+        Html::addCssClass($options, $this->errorSummaryCssClass);
+
+        $options['encode'] = $this->encodeErrorSummary;
+
+        return ErrorSummary::widget()->form($forms)->options($options)->run();
```

Comment:
I don't see `form()` method in `ErrorSummary`.

Reply:
remove.